cli/connhelper/commandconn/commandconn.go

```diff
@@ -178,9 +177,8 @@ func (c *commandConn) Write(p []byte) (int, error) {
 	// Close might get called
 	if c.closing.Load() {
 		// If we're currently closing the connection
-		// we don't want to call onEOF, but we do want
-		// to return an io.EOF
-		return 0, io.EOF
+		// we don't want to call onEOF
+		return n, err
```

Comment:
Err may be `nil` here; do we need to return a `io.EOF` in that case? (Thinking out loud if we need an `io.EOF` for calling code)

```go
if err == nil {
    return io.EOF
}
```

Reply:
No, if the read didn't return an error then it's fine to not return one here. This was the original behavior, and subsequent reads will fail regardless with a file closed/pipe closed error.

Reply:
In normal operation, it will only be closing if an EOF was already returned triggering the transport layer to call Close regardless. This special handling is only meant to keep us from calling handleEOF twice.

Reply:
Thanks for looking! The original comment triggered my curiosity, so wanted to check 😅  👍